components/gitpod-db/src/typeorm/webhook-event-db-impl.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { injectable, inject } from "inversify";
+import { Repository } from "typeorm";
+import { v4 as uuidv4 } from "uuid";
+
+import { TypeORM } from "./typeorm";
+import { WebhookEventDB } from "../webhook-event-db";
+import { DBWebhookEvent } from "./entity/db-webhook-event";
+import { WebhookEvent } from "@gitpod/gitpod-protocol";
+
+@injectable()
+export class WebhookEventDBImpl implements WebhookEventDB {
+    @inject(TypeORM) protected readonly typeORM: TypeORM;
+
+    protected async getEntityManager() {
+        return (await this.typeORM.getConnection()).manager;
+    }
+
+    protected async getRepo(): Promise<Repository<DBWebhookEvent>> {
+        return (await this.getEntityManager()).getRepository(DBWebhookEvent);
+    }
+
+    async createEvent(parts: Omit<WebhookEvent, "id">): Promise<WebhookEvent> {
+        const repo = await this.getRepo();
+        const newEvent: WebhookEvent = {
+            ...parts,
+            id: uuidv4(),
+            creationTime: new Date().toISOString(),
+        };
+        return await repo.save(newEvent);
+    }
+
+    async updateEvent(id: string, update: Partial<WebhookEvent>): Promise<void> {
+        const repo = await this.getRepo();
+        const safeUpdate: Partial<WebhookEvent> = { ...update, id };
+        delete safeUpdate.type;
+        delete safeUpdate.rawEvent;
+        await repo.save(safeUpdate);
+    }
+
+    async find(filter: { cloneUrl?: string | undefined; projectId?: string | undefined }): Promise<WebhookEvent[]> {
+        const repo = await this.getRepo();
+        if (!filter.cloneUrl && !filter.projectId) {
+            return [];
+        }
+        const query = repo.createQueryBuilder("event");
+        if (filter.projectId) {
+            query.where("event.projectId = :projectId", { projectId: filter.projectId });
+        } else if (filter.cloneUrl) {
```

Comment:
If this really is exclusive I think we should have two methods: `findByProjectId` and `findByCloneUrl`.

:cloud: For both queries we should have indexes: `ind_projectId_creationTime` and `ind_cloneUrl_creationTime`

Reply:
Agree and introduce `findByProjectId` only once we need it.